lib/dash/dash_parser.js

```diff
@@ -328,8 +328,13 @@ shaka.dash.DashParser = class {
           mpd, 'suggestedPresentationDelay', XmlUtils.parseDuration);
     }
 
-    const maxSegmentDuration = XmlUtils.parseAttr(
-        mpd, 'maxSegmentDuration', XmlUtils.parseDuration);
+    const ignoreMaxSegmentDuration = this.config_.dash.ignoreMaxSegmentDuration;
+    let maxSegmentDuration = 1;
+    if (!ignoreMaxSegmentDuration) {
+      maxSegmentDuration = XmlUtils.parseAttr(
```

Comment:
The `|| 1` in the original version was to provide a default if the property does not exist.  `parseAttr` returns `null` in this case.  So you'll still need to fall back to 1 somewhere if this branch is taken and it returns null.  You could just add the `|| 1` back to the `notifyMaxSegmentDuration` call below, or you could provide a default value as another argument to `parseAttr`:

```js
      maxSegmentDuration = XmlUtils.parseAttr(
          mpd, 'maxSegmentDuration', XmlUtils.parseDuration,
          /* defaultValue= */ 1);
```